### PR TITLE
`azurerm_automation_account`  - support for the `encryption` and `local_authentication_enabled` property

### PR DESCRIPTION
**Pull request by wuxu92**
support of `encryption` and `disable_local_auth` 

--- PASS: TestAccAutomationAccount_basic (119.18s)
--- PASS: TestAccAutomationAccount_requiresImport (128.41s)
--- PASS: TestAccAutomationAccount_complete (66.28s)
--- PASS: TestAccAutomationAccount_encryption (298.88s)
--- PASS: TestAccAutomationAccount_identityUpdate (157.56s)
--- PASS: TestAccAutomationAccount_systemAssignedIdentity (68.73s)
--- PASS: TestAccAutomationAccount_systemAssignedUserAssignedIdentity (79.94s)
--- PASS: TestAccAutomationAccount_userAssignedIdentity (145.22s)
PASS

### CODE REVIEW

File: internal/services/automation/automation_account_resource.go [hunk 1]
```diff
@@ -58,6 +60,50 @@ func resourceAutomationAccount() *pluginsdk.Resource {
 
 			"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
 
+			"encryption": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_identity_id": {
```
Comment (stephybun) [inline internal/services/automation/automation_account_resource.go:69]: Can we rename this to be consistent with other encryption blocks within the provider?
```suggestion
						"user_assigned_identity_id": {
```

File: website/docs/r/automation_account.html.markdown [hunk 2]
```diff
@@ -60,6 +64,20 @@ An `identity` block supports the following:
 
 -> **Note:** `identity_ids` is required when `type` is set to `UserAssigned` or `SystemAssigned, UserAssigned`.
 
+--
+
+An `encryption` block supports the following:
+
+* `user_identity_id` - (Optional) The user identity used for CMK. It will be an ARM resource id.
```
Comment (stephybun) [inline website/docs/r/automation_account.html.markdown:71]:
```suggestion
* `user_identity_id` - (Optional) The User Assigned Managed Identity ID to be used for accessing the Customer Managed Key for encryption.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this @wuxu92, this is off to a solid start. I left some suggestions in-line, once those are fixed up we can take another look through.

File: internal/services/automation/automation_account_resource.go [hunk 3]
```diff
@@ -58,6 +60,50 @@ func resourceAutomationAccount() *pluginsdk.Resource {
...
+						},
+						"key_version": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"key_vault_uri": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
```
[the remark this answers is not in the document]
Comment (wuxu92) [reply internal/services/automation/automation_account_resource.go:96]: Is that mean we do a query to get the `key_name`, `key_version`, and `key_vault_uri` by the `key_vault_key_id` and then fill them into the request model?  looks like no common way to do this in our repo.

[the remark this answers is not in the document]
Comment (stephybun) [reply internal/services/automation/automation_account_resource.go:96]: Here is an example of what I mean https://github.com/hashicorp/terraform-provider-azurerm/blob/4620cb506b866b649c7f666e54ba2431825a2475/internal/services/cognitive/cognitive_account_customer_managed_key_resource.go#L96-L104

**Comment by wuxu92** [comment]
Hi @stephybun i have updated the pr to fix the comments could you please have another look, thanks!

File: internal/services/automation/automation_account_resource.go [hunk 4]
```diff
@@ -58,6 +62,41 @@ func resourceAutomationAccount() *pluginsdk.Resource {
...
+				Optional: true,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_assigned_identity_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: commonids.ValidateUserAssignedIdentityID,
+						},
+						"key_source": {
```
Comment (katbyte) [inline internal/services/automation/automation_account_resource.go:76]: can we space things consistanty with the rest of the provider
```suggestion
						},

						"key_source": {
```

File: internal/services/automation/automation_account_resource.go [hunk 5]
```diff
@@ -58,6 +62,41 @@ func resourceAutomationAccount() *pluginsdk.Resource {
...
+						"key_vault_key_id": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: keyVaultValidate.NestedItemIdWithOptionalVersion,
+						},
+					},
+				},
+			},
+
+			"local_auth_enabled": {
```
Comment (katbyte) [inline internal/services/automation/automation_account_resource.go:94]:
```suggestion
			"local_authentication_enabled": {
```

File: internal/services/automation/automation_account_resource_test.go [hunk 6]
```diff
@@ -257,6 +274,121 @@ resource "azurerm_automation_account" "test" {
...
+  sku_name            = "Basic"
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.test.id
+    ]
+  }
+  local_auth_enabled = false
+  encryption {
```
Comment (katbyte) [inline internal/services/automation/automation_account_resource_test.go:383]:
```suggestion
  }
  
  local_auth_enabled = false
  
  encryption {
```

File: website/docs/r/automation_account.html.markdown [hunk 7]
```diff
@@ -44,12 +44,16 @@ The following arguments are supported:
 
 * `sku_name` - (Required) The SKU of the account - only `Basic` is supported at this time.
 
+* `disable_local_auth` - (Optional) Whether requests using non-AAD authentication are blocked.
```
Comment (katbyte) [inline website/docs/r/automation_account.html.markdown:47]: this hasn't been updated?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by wuxu92** [comment]
@katbyte pr updated and all checks passed now

**Review by katbyte — APPROVED** [review]
LGTM 🔒

**Pull request merged**